packages/clay-tabs/src/Content.tsx

```diff
@@ -29,13 +34,9 @@ const Content: React.FunctionComponent<IProps> = ({
 	return (
 		<div className={classNames('tab-content', className)} {...otherProps}>
 			{React.Children.map(children, (child, index) => {
-				if (!React.isValidElement(child)) {
-					return null;
-				}
-
 				return React.cloneElement(child, {
 					...child.props,
-					active: activeTabKey === child.props.tabKey,
+					active: activeTabKey === child.props.tabkey,
```

Comment:
I feel a bit queasy about accessing `child.props` from the outside. The fact that the `TabPane` component takes `tabkey` as a prop but doesn't actually access it is another sign that this is a smell.

This feels like the kind of thing that the component should take care of instead of the user; ie. in the example below, having to provide a `tabkey` is just expanding with surface area of the API with another unique prop (like `tabName`), which is another thing for me to have to worry about and possibly get wrong. It feels a bit like a leaky abstraction; we're exposing what would ideally be internal-only plumbing:

```javascript
<ClayTabs>
	<ClayTabs.Item
		active={activeTabKeyValue == 0}
		onClick={onClick}
		tabName="Dummy1"
		tabkey={0}
	/>
	<ClayTabs.Item
		active={activeTabKeyValue == 1}
		data-testid="tabItem2"
		onClick={onClick}
		tabName="Dummy2"
		tabkey={1}
	/>
	<ClayTabs.Item
		active={activeTabKeyValue == 2}
		onClick={onClick}
		tabName="Dummy3"
		tabkey={2}
	/>
</ClayTabs>
```

it would be nice if we could take care of that. I have a feeling that could be made to work; did you explore that in the initial implementation?

Reply:
Yes, Initially I implemented using the order of the `Item`s that matches with the order of `Content` but I wasn't able to use Dropdown using this logic.

Personally, I prefer this version of the [API](https://github.com/diegonvs/clay/blob/47526867448545c7372466cb55e58a1e0cab4f0f/packages/clay-tabs/stories/index.tsx), however, I don't know what is the number of the items that I need for creating a [dropdown](https://issues.liferay.com/browse/LEXI-664) and I was thinking about this API could be high-level. 

Check out the versions that I tried before here: https://github.com/diegonvs/clay/commits/tabs